### PR TITLE
Add AQ3DC module

### PR DESCRIPTION
**Pull request by baptistebaquero**
This Slicer module's aim is to automatically compute head measurements used in craniofacial surgery. Using placed fiducials, allows users to compute 2D angles: Yaw, Pitch, and Roll; and decompose the 3D distance (between points, points to line) into the three different components: R-L, A-P, and S-I. It is possible to compute the middle point between two fiducials and export the values. It is possible to compute lists of measurements selected by users for a single case or a whole study sample, at one or more time points.
There is clinicians need to compute different measurements for a whole sample of patients instead of doing each measurement one by one for each patient.

### CODE REVIEW

File: AQ3DC/AQ3DC.py [hunk 1]
```diff
@@ -0,0 +1,1062 @@
...
+  # print(list_label)
+  return dic_tooth
+
+DistanceResult = collections.namedtuple("DistanceResult", ("delta", "norm"))
+def computeDistance(point1, point2) -> DistanceResult:
+    delta = np.abs(np.subtract(point2,point1))
+    return DistanceResult(
+        delta,
+        np.linalg.norm(delta),
+    )
```
Comment (allemangD) [inline AQ3DC/AQ3DC.py:175]: This can be imported from the `DependantMarkups` module.

```python3
from DependantMarkups import DependantMarkupsLogic
```

You should then have an instance of this as an attribute in `AQ3DCLogic`:

```python3
  def __init__(self):
    self.dependencies = DependantMarkupsLogic()
```

You can then use methods like `self.dependencies.computeDistance` or `self.dependencies.setProjected` as described in [/DependantMarkups/README.md](https://github.com/DCBIA-OrthoLab/Q3DCExtension/blob/dependant-markups/DependantMarkups/README.md)

**Review by pieper — APPROVED** [review]
Much better - thank you!

**Comment by jcfr** [comment]
Thanks for the updates :pray: 

Here are few additional comment:
* [ ] Provide an icon
* [ ] Update top-level `README` to reference both `Q3DC` and `AQ3DC`, briefly explain why the user would choose one vs the other

File: AQ3DC/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# AQ3DC : Automatic Quantification for 3D Cephalometrics
+
+Authors: Baptiste Baquero, Maxime Gillot, Lucia Cevidanes, Jean-Christophe Fillion-Robin, David Allemang
+	
+## What is it?
+
+This Slicer extension aims is to automatically computes head measurements used in craniofacial surgery. Using placed fiducials, it allows users to compute 2D angles: Yaw, Pitch and Roll; and decompose the 3D distance (between points, points to line) into the three different components: R-L , A-P and S-I. It is possible to compute the middle point between two fiducials and export the values. It is possible to compute lists of measurements selected by users for a single case or a whole study sample, at one or more time points.
```
Comment (jcfr) [inline AQ3DC/README.md:7]: Update wording replacing `extension` with `module`

File: AQ3DC/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ 
+## File Formats
+
+AQ3DC allows users to compute all the measurements on points from .json files. Save middle points created in a .json file. Measurement list and results of all the computed measurements saved as .xlsx files.
```
Comment (jcfr) [inline AQ3DC/README.md:11]: > from .json files

The format of the input file should be documented

File: AQ3DC/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+## Prerequisites for users
+
+The AQ3DC extension really need consistency in the data. For each patient, in the same time point and at different time point, if the landmarks are not named exactly the same the measurement for this patient won't be perform, and will just be skiped.
```
Comment (jcfr) [inline AQ3DC/README.md:15]: Is this sentence related to the input json file ?

**Review by jcfr — COMMENTED** [review]
Thanks for the updates :pray: 

Here are few additional comment:
* [ ] Provide an icon for the `AQ3DC` module
* [ ] Use sections like `Panels and their use` and `Use cases` for each module (see example in Slicer at [Docs/user_guide/modules](https://github.com/Slicer/Slicer/tree/main/Docs/user_guide/modules)). For example, see [transforms](https://github.com/Slicer/Slicer/blob/main/Docs/user_guide/modules/transforms.md). Browsing the [Slicer modules documentation](https://slicer.readthedocs.io/en/latest/user_guide/modules/index.html) will help find relevant examples.
* [ ] Update top-level `README`
  * reference both `Q3DC` and `AQ3DC`, briefly explain why the user would choose one vs the other.
  * consider merging all information into the top-level README. Currently, there are `Q3DCExtension/README.md` and `Q3DCExtension/AQ3DC/README.md` but there is no `README.md` for `Q3DC`.

**Comment by baptistebaquero** [comment]
> Thanks for the updates pray
> 
> Here are few additional comment:
> 
>     * [ ]  Update pull request description to provide a message summarizing the contribution. This will be used when squashing the commits. It should include a high level overview of what the module does, its features, motivation and how it compares with the existing Q#DC module.
> 
>     * [ ]  Provide an icon for the `AQ3DC` module
> 
>     * [ ]  README: Add a description of the format for the input json file along with a simple example.
> 
>     * [ ]  README: Ideally use sections like `Panels and their use` and `Use cases` (see example in Slicer at [Docs/user_guide/modules](https://github.com/Slicer/Slicer/tree/main/Docs/user_guide/modules)). For example, see [transforms](https://github.com/Slicer/Slicer/blob/main/Docs/user_guide/modules/transforms.md). Browsing the [Slicer modules documentation](https://slicer.readthedocs.io/en/latest/user_guide/modules/index.html) will help find relevant examples.
> 
>     * [ ]  README: Add reference to sample data for testing the module
> 
>     * [ ]  Update top-level `README`
>       
>       * reference both `Q3DC` and `AQ3DC`, briefly explain why the user would choose one vs the other.
>       * consider merging all information into the top-level README. Currently, there are `Q3DCExtension/README.md` and `Q3DCExtension/AQ3DC/README.md` but there is no `README.md` for `Q3DC`.
>       * consider looking at this README for an example:
>         https://github.com/KitwareMedical/SlicerPipelines/blob/main/README.md
> 
>     * README: Consider adding an history section summarizing when/who/... the Q3DC and AQ3DC modules where developed. For an example of history sections, see:
>       
>       * https://github.com/Slicer/SlicerSurfaceToolbox#history
>       * https://github.com/Slicer/SlicerDataStore#history
>       * https://github.com/Slicer/Slicer-DataStore#history

**Comment by baptistebaquero** [comment]
Hello JC and David,
- I provided a Icon for AQ3DC called AQ3DC.png .
- I added some json files in the folder "docs" to test the module.
- I created a "Panels and their use", I tried to put screenshots with short description of each part in the module, tell me what do you think about it.
- There is now just one README for Q3DC and AQ3DC.
- I created an History but I'm not sure about the year of creation of Q3DC.

Tell me if there is still issues to fix.

File: AQ3DC/README.md (hunk 4 above)
Comment (allemangD) [reply AQ3DC/README.md]: The JSON files are standard `.mrk.json` landmark files. This is referring to the fact that AQ3DC uses simple string comparison for measurements; the landmark label must be an exact match to the point for measurement, otherwise the measurement will be skipped.

**Comment by allemangD** [comment]
Squashed commits for more meaningful history, and addressed those few remaining issues pointed out by @jcfr. Merging now.

Thanks for all the great work @baptistebaquero!
